core/iwasm/include/aot_comp_option.h

```diff
@@ -6,6 +6,26 @@
 #ifndef __AOT_COMP_OPTION_H__
 #define __AOT_COMP_OPTION_H__
 
+typedef struct {
+    /* Enables or disables bounds checks for stack frames. When enabled, the AOT
+     * compiler generates code to check if the stack pointer is within the
+     * bounds of the current stack frame (and if not, traps). */
+    bool bounds_checks;
+
+    /*  Enables or disables instruction pointer (IP) tracking.*/
+    bool ip;
+
+    /* Enables or disables tracking instruction pointer of a trap. Only takes
+     * effect when `ip` is enabled.*/
+    bool trap_ip;
+
+    /* Enables or disables stack pointer (SP) tracking. */
+    bool sp;
+
+    /* Enables or disables imported functions' parameters tracking. */
+    bool params;
```

Comment:
Several questions about the options:
(1) the func_idx will be always committed so no need to add an option for it, right?

(2) in fact, not only function params can be committed, but also function's locals and stack operands. Now they are now committed only when GC feature is enabled: `aot_gen_commit_values()` is called with `if (comp_ctx->enable_gc
                && !aot_gen_commit_values(comp_ctx->aot_frame))` like code.
do you think that we should enable committing them and change `params` to other word like `values`?

(3) by default, GC should enable sp and params (and locals, stack operands), or AOT runtime may run failed. So had better set them true when GC is enabled, no matter what user enables them or not in this flag?

Reply:
> Several questions about the options: (1) the func_idx will be always committed so no need to add an option for it, right?
> 

Yes, I don't know if there's any usecase to disable `func_idx`; if needed, that option can be added in the future though.

> (2) in fact, not only function params can be committed, but also function's locals and stack operands. Now they are now committed only when GC feature is enabled: `aot_gen_commit_values()` is called with `if (comp_ctx->enable_gc && !aot_gen_commit_values(comp_ctx->aot_frame))` like code. do you think that we should enable committing them and change `params` to other word like `values`?
> 

I think we could rename it to values and cover more data; if needed this can be more granular in the future

> (3) by default, GC should enable sp and params (and locals, stack operands), or AOT runtime may run failed. So had better set them true when GC is enabled, no matter what user enables them or not in this flag?

isn't stack pointer and stack operands only actually used in the GC mode? If so, should the SP option be deleted at all?

Reply:
> > Several questions about the options: (1) the func_idx will be always committed so no need to add an option for it, right?
> 
> Yes, I don't know if there's any usecase to disable `func_idx`; if needed, that option can be added in the future though.
> 

OK, sounds good.

> > (2) in fact, not only function params can be committed, but also function's locals and stack operands. Now they are now committed only when GC feature is enabled: `aot_gen_commit_values()` is called with `if (comp_ctx->enable_gc && !aot_gen_commit_values(comp_ctx->aot_frame))` like code. do you think that we should enable committing them and change `params` to other word like `values`?
> 
> I think we could rename it to values and cover more data; if needed this can be more granular in the future
> 

Yes, agree to rename to values.

> > (3) by default, GC should enable sp and params (and locals, stack operands), or AOT runtime may run failed. So had better set them true when GC is enabled, no matter what user enables them or not in this flag?
> 
> Good to know. But isn't stack pointer and stack operands only actually used in the GC mode? If so, should the SP option be deleted at all?

Yes, normally sp and stack operands should be committed at the same time, or it is meaningless for the runtime, and now they are really only used in the GC mode. Agree to remove sp flag and commit sp when committing `values` is enabled, and set `values` to true when GC is enabled. I am not sure whether in the future, the stack operands will be used in other features, like a break point in AOT debugger to view the locals and stacks operands, or dumping more details when an exception is thrown, so had better also keep values flag in `--call-stack-features=`?

Reply:
Agree, I'll make the changes once those two PRs (https://github.com/bytecodealliance/wasm-micro-runtime/pull/3768, https://github.com/bytecodealliance/wasm-micro-runtime/pull/3767) to avoid some potential merge conflicts.

Reply:
@wenyongh I just made an update, please let me know if you have any feedback.

Reply:
It is good to me. I am not sure whether others have comments, let's merge it late today or early tomorrow if there is no new comment?

Reply:
sounds good. Once this is merged, I'll open another CR with a tiny stack frame then.